### PR TITLE
RSpec::Support::FuzzyMatcher.failure_message implemented…

### PR DESCRIPTION
**Pull request by faucct**
… for providing informative failure messages

### CODE REVIEW

**Comment by myronmarston** [comment]
What are you trying to achieve here?  `FuzzyMatcher` is not an official rspec-expectations matcher, and isn't meant to be used directly by end users.  Can you show an example of what you wanted to do (but could not) that this PR would fix?

**Comment by faucct** [comment]
Hello, I want to make the rspec-core's `include_attributes` failure_message more descriptive and it utilizes the `FuzzyMatcher`.

**Comment by myronmarston** [comment]
Sounds like the same issue of rspec/rspec-expectations#759.  Does that sound right?

Regardless, any improvements to the `have_attributes` matcher are going to have to be made in rspec-expectations, since that's where it lives.   `FuzzyMatcher` can expose some additional APIs to support corresponding changes over there as needed of course.  Can you open a PR showing what you have in mind for the changes to `have_attributes`?

File: lib/rspec/support/fuzzy_matcher.rb [hunk 1]
```diff
@@ -3,46 +3,100 @@ module Support
...
 
-        return true if expected == actual
+        return if expected == actual
 
         begin
-          expected === actual
+          if expected === actual
+            return
+          else
+            if Matchers::BuiltIn::BaseMatcher === expected
```
Comment (myronmarston) [inline lib/rspec/support/fuzzy_matcher.rb]: RSpec::Support must be usable w/o rspec-expectations loaded (which is the gem that defines `Matchers::BuiltIn::BaseMatcher`) so doing this kind of check here is problematic.

**Comment by faucct** [comment]
I am back with cleaned up code. The `have_attributes` PR example will take some time, I will do it a little bit later.

**Comment by JonRowe** [comment]
:wave: I'm closing this PR as stale for now, it also makes too many breaking changes to `FuzzyMatcher` any improvements really need to be focused on the matcher in question. You certainly can't change this to a class.

**Pull request closed without merging**